tests/src/main/java/testcases/T057_generic_dynamic_dependency/Child.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package testcases.T057_generic_dynamic_dependency;
+
+import motif.Scope;
+
+import java.util.function.Supplier;
+
+@Scope
+public interface Child {
+
+    String string();
+
+    @motif.Objects
+    class Objects {
+
+        String string(Supplier<String> supplier) {
```

Comment:
what about nested cases like `List<Map<String, Integer>>`?

Reply:
As discussed offline: Will leave as is since we're using JavaPoet under the hood and it should handle these cases already. In other words, we don't want to duplicate JavaPoet's test coverage here.